mono/metadata/class-init.c

```diff
@@ -2256,7 +2256,13 @@ mono_class_layout_fields (MonoClass *klass, int base_instance_size, int packing_
 		klass->instance_size = instance_size;
 	}
 	klass->blittable = blittable;
-	klass->has_references = has_references;
+	/* An Ephemeron cannot be marked by Boehm */
+	/* See SGen equivalent code in compute_class_bitmap */
+	if (!mono_gc_is_moving() && m_class_get_image (klass) == mono_defaults.corlib && !strcmp ("Ephemeron", m_class_get_name (klass))) {
```

Comment:
Nit: I would prefer `strcmp ("Ephemeron", m_class_get_name (klass)) == 0` here, since the `!` makes me think "not equal" initially. But we can match the code style in this file already.

Reply:
This was a copy paste from the Mono sgen version of this. i can change it if I push other updates.

Reply:
Ok, no big deal. It is probably better to leave the code as-is then.